source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+   :end-before: # end-custom-polymorphic
+
+In the preceding example, the ``identify_as`` directive instructs {+odm+}
+to store the ``Band`` class in the database as the string ``"artist"``. 
+
+You can also specify multiple aliases, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-multiple-alias
+   :end-before: # end-multiple-alias
+
+In the preceding example, ``artist`` is the default name and the others are used
+only for looking up records. This allows you to refactor your
+code without breaking the associations in your data. 
+
+Polymorphic type aliases are global. The keys you specify must be unique across your
+entire code base. However, it is possible to register alternative resolvers that
+can be used for different subsets of your models. In this case, the keys must
+only be unique for each resolver. The following example shows how to register
+alternate resolvers:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-polymorphic-resolvers
+   :end-before: # end-polymorphic-resolvers
+
+Both ``Music::Band`` and ``Tools::Band`` are aliased as 
+``"bnd"``, but use their own resolver to avoid conflicts.
+
+Dependent Behavior
+~~~~~~~~~~~~~~~~~~
+
+You can provide ``dependent`` options to referenced associations to specify how
+{+odm+} handles associated documents when a document is deleted. You can specify
+the following options:
+
+- ``delete_all``: Deletes all child documents without running any model
+  callbacks.
+- ``destroy``: Deletes the child documents and runs all model callbacks.
+- ``nullify``: Sets the foreign key of the child documents to ``nil``. The child
+  document might become orphaned if it is referenced by only the parent.
+- ``restrict_with_exception``: Raises an exception if the child document is not
+  empty.
+- ``restrict_with_error``: Cancels the operation and returns ``false`` if the
+  child document is not empty.
+
+If you don't specify any ``dependent`` options, {+odm+} leaves the child
+document unchanged when the parent document is deleted. The child document
+continues to reference the deleted parent document. If the child document is
+referenced through only the parent, the child document becomes orphaned.
+
+The following example specifies ``dependent`` options on the ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-dependent
+   :end-before: # end-dependent
+
+Autosave Referenced Associations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, {+odm+} does not automatically save associated documents from
+non-embedded associations when saving the parent document. Because of this, it's
+possible to create dangling references to documents that don't exist through
+associations. You can use the ``autosave`` option on a referenced association to
```

Comment:
s:

```suggestion
non-embedded associations when saving the parent document. This can
result in dangling references to documents that don't exist.
 You can use the ``autosave`` option on a referenced association to
```